stwcs/tests/test_altwcs.py

```diff
@@ -169,3 +172,103 @@ def test_pars_wcskey(self):
         #assert(altwcs._parpasscheck(f, ext=1, wcskey='O'))
         #assert(not altwcs._parpasscheck(f, ext=1, wcskey='O', reusekey=False))
         f.close()
+
+    def _prepare_acs_test_file(self, ext_list):
+        shutil.copyfile(self.acs_orig_file, self.acs_file)
+        idctab = get_filepath('postsm4_idc.fits')
+        npol_file = get_filepath('qbu16424j_npl.fits')
+        d2imfile = get_filepath('new_wfc_d2i.fits ')
+        pyfits.setval(self.acs_file, ext=0, keyword="IDCTAB", value=idctab)
+        pyfits.setval(self.acs_file, ext=0, keyword="NPOLFILE", value=npol_file)
+        pyfits.setval(self.acs_file, ext=0, keyword="D2IMFILE", value=d2imfile)
+
+        h = pyfits.open(self.acs_file, mode='update')
+
+        # remove all Alt WCS except OPUS:
+        keys = altwcs.wcskeys(h, 1)
+        if ' ' in keys:
+            keys.remove(' ')
+        if 'O' in keys:
+            keys.remove('O')
+
+        for n, ext in enumerate(ext_list):
+            for key in keys:
+                altwcs.deleteWCS(h, ext, wcskey=key)
+
+                # remove HDRNAME:
+                if ('HDRNAME' + key) in h[ext].header:
+                    del h[ext].header['HDRNAME' + key]
+
+            h[ext].header['wcsname'] = 'Initial Primary WCS'
+            h[ext].header['CD1_1'] = n + 1.234
+            # remove HDRNAME:
+            if 'HDRNAME' in h[ext].header:
+                del h[ext].header['HDRNAME']
+
+        envval = os.environ['ASTROMETRY_STEP_CONTROL']
+        os.environ['ASTROMETRY_STEP_CONTROL'] = 'Off'
+        updatewcs.updatewcs(h, use_db=False)
+        os.environ['ASTROMETRY_STEP_CONTROL'] = envval
+
+        h.close()
+
+    def test_repeated_updatewcs_no_db(self):
+        ext_list = [('sci', 1), ('sci', 2)]
+
+        # remove all Alt WCS except OPUS:
+        self._prepare_acs_test_file(ext_list)
+
+        # run updatewcs() multiple times:
+        for k in range(4):
+            updatewcs.updatewcs(self.acs_file, use_db=False)
+
+        # test:
+        h = pyfits.open(self.acs_file, mode='update')
```

Comment:
This decision comes from https://github.com/spacetelescope/stwcs/blob/831ea8a5ba0e78fb668a792ecccca0321eb41df1/stwcs/wcsutil/altwcs.py#L836-L839 but I do not think this should be a requirement. I understand where this comes from but I think once an `HDUList` is loaded from a file even in read only mode, it can be saved later with `writeto('a_different_file.fits')`. So, if everyone is OK, I'll be happy to remove this restriction.

Reply:
Ah, yes, This requirement came from the desire to process the files in memory on AWS without downloading them to disk first. Let's not deal with this now.

Reply:
But I do not think removing this requirement would hurt anything that was designed to work with `'update'` mode ON - it is caller's responsibility to use code responsibly :)